src/detect-engine-hsbd.c

```diff
@@ -263,9 +267,31 @@ int DetectEngineRunHttpServerBodyMpm(DetectEngineCtx *de_ctx,
     if (buffer_len == 0)
         goto end;
 
+    if (htp_state->cfg->decomp_swf_enabled) {
+        int file_swf_type = FileIsFlashFile(buffer, buffer_len);
+        if (file_swf_type == FILE_FLASH_ZLIB_COMPRESSION ||
+            file_swf_type == FILE_FLASH_LZMA_COMPRESSION)
+        {
+            ret = FileDecompressFlashFile(buffer, buffer_len,
+                                          &decompressed_buffer,
+                                          &decompressed_buffer_len,
+                                          det_ctx,
+                                          htp_state->cfg->decomp_swf_type,
+                                          htp_state->cfg->decomp_swf_decompress_depth,
+                                          htp_state->cfg->decomp_swf_compress_depth);
+            if (ret == 1) {
+                buffer = decompressed_buffer;
+            }
+        }
+    }
+
     cnt = HttpServerBodyPatternSearch(det_ctx, (uint8_t *)buffer, buffer_len, flags);
 
  end:
+    if (ret == 1) {
+        SCFree(decompressed_buffer);
```

Comment:
Both Mpm and Inspect engine free their mem after use, which means we're doing the same work twice. If mpm matches, the inspect engine (DetectEngineInspectHttpServerBody) will be called which has to uncompress the same file again?